tests/Unit/NumericalAlgorithms/LinearOperators/Test_PartialDerivatives.cpp

```diff
@@ -576,6 +578,74 @@ void test_partial_derivatives_3d(const Mesh<3>& mesh) {
     }
   }
 }
+
+template <typename VariableTags, typename GradientTags = VariableTags>
+void test_partial_derivatives_spherical_shell() {
+  const size_t L = 4;
+  const size_t n_r = 4;
+  const Mesh<3> mesh{
+      {n_r, L + 1, 2 * L + 1},
+      {Spectral::Basis::Legendre, Spectral::Basis::SphericalHarmonic,
+       Spectral::Basis::SphericalHarmonic},
+      {Spectral::Quadrature::Gauss, Spectral::Quadrature::Gauss,
+       Spectral::Quadrature::Equiangular}};
+  const size_t number_of_grid_points = mesh.number_of_grid_points();
+  const auto prod_map3d =
+      domain::make_coordinate_map<Frame::ElementLogical, Frame::Grid>(
+          domain::CoordinateMaps::ProductOf2Maps<
+              Affine, domain::CoordinateMaps::Identity<2>>{
+              Affine{-1.0, 1.0, 1.0, 1.5},
+              domain::CoordinateMaps::Identity<2>{}},
+          domain::CoordinateMaps::SphericalToCartesianPfaffian{});
+  const auto xi = logical_coordinates(mesh);
+  const auto x = prod_map3d(xi);
+  const InverseJacobian<DataVector, 3, Frame::ElementLogical, Frame::Grid>
+      inverse_jacobian = prod_map3d.inv_jacobian(xi);
+
+  Variables<VariableTags> u(number_of_grid_points);
+  Variables<
+      db::wrap_tags_in<Tags::deriv, GradientTags, tmpl::size_t<3>, Frame::Grid>>
+      expected_du(number_of_grid_points);
+  Approx local_approx = Approx::custom().epsilon(1e-9).scale(1.0);
```

Comment:
Do you need to force the scale to 1 instead of just a relative error? Also, 1e-9 seems sort of large? Do you know why it's that large?

Reply:
- yes, because some of the derivs are zero.
- because I copied it from the generic 3d test.  I've updated it to 1e-13.